File: backend/app/schemas/template_reviewer.py
"""
Pydantic schemas for TemplateReviewer resources.

Reviewers are specified as a list when the template is created.
Like columns, reviewers are managed as a batch.
"""

from uuid import UUID

from pydantic import BaseModel, EmailStr, Field

from app.schemas.common import ORMBase


class TemplateReviewerCreate(BaseModel):
    """
    Request body for adding a single reviewer to a template.

    POST /templates/{id}/reviewers accepts a list of these.
    """

    reviewer_email: EmailStr = Field(
        ...,
        description="Reviewer's email address. They will receive approval request emails here."
    )
    reviewer_name: str = Field(
        ...,
        min_length=1,
        max_length=100,
        description="Reviewer's display name."
    )
    reviewer_type: str = Field(
        'required',
        description="Required reviewers must all approve. Optional reviewers can see the request but approval is not blocking."
    )


class TemplateReviewerResponse(ORMBase):
    """Response shape for template reviewers."""

    id: UUID
    template_id: UUID
    reviewer_email: str
    reviewer_name: str
    reviewer_type: str

File: frontend/src/types/template_reviewer.ts
/**
 * TypeScript types for the TemplateReviewer resource.
 *
 * Mirrors backend/app/schemas/template_reviewer.py.
 *
 * A reviewer is required or optional. Required reviewers must
 * all approve before a template is activated.
 */

export type ReviewerType = 'required' | 'optional'

export interface TemplateReviewer {
  id: string
  template_id: string

  reviewer_email: string
  reviewer_name: string
  reviewer_type: ReviewerType
}

export interface TemplateReviewerCreate {
  reviewer_email: string
  reviewer_name: string
  reviewer_type?: ReviewerType
}